app/lib/__init__.py

```diff
@@ -1 +1,99 @@
+from datetime import date
+from dateutil.relativedelta import MO, TH
+from dateutil.relativedelta import relativedelta as rd
+from holidays import HolidayBase
 
+
+class NYCHolidays(HolidayBase):
+    def __init__(self, **kwargs):
+        self.country = 'US'
+        HolidayBase.__init__(self, **kwargs)
+
+    def _populate(self, year):
+        # New Year's Day
+        if year > 1870:
+            name = "New Year's Day"
+            self[date(year, 1, 1)] = name
+            if self.observed and date(year, 1, 1).weekday() == 6:
+                self[date(year, 1, 1) + rd(days=+1)] = name + " (Observed)"
+            elif self.observed and date(year, 1, 1).weekday() == 5:
+                # Add Dec 31st from the previous year without triggering
+                # the entire year to be added
+                expand = self.expand
```

Comment:
Joel says you can ignore this or try and change it and see if it breaks anything.